From 789e22260b121f9f5c20361b3410754cca1501bf Mon Sep 17 00:00:00 2001
From: Johnny Mnemonic <jm@machine-hall.org>
Date: Mon, 27 Jan 2025 18:30:30 +0100
Subject: [PATCH 56/96] ia64: Make expand_upwards() available again

"mm/vma: move stack expansion logic to mm/vma.c" (commit a9d1f3f2d7fecbc465bd4b16343a2ff8499cc558)
didn't anticipate that ia64 also needs this when putting a guard around
the function prototype for it in mm/vma.h. This broke the build for ia64:

```
mm/mmap.c: In function 'vma_expand_up':
mm/mmap.c:1033:16: error: implicit declaration of function 'expand_upwards'; did you mean 'expand_downwards'? [-Wimplicit-function-declaration]
 1033 |         return expand_upwards(vma, addr);
      |                ^~~~~~~~~~~~~~
      |                expand_downwards
```

...and is fixed now.
---
 mm/vma.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mm/vma.h b/mm/vma.h
index 9d5ee6ac913a..83353f64762e 100644
--- a/mm/vma.h
+++ b/mm/vma.h
@@ -613,7 +613,7 @@ static inline bool vma_is_sealed(struct vm_area_struct *vma)
 }
 #endif
 
-#if defined(CONFIG_STACK_GROWSUP)
+#if ( defined(CONFIG_STACK_GROWSUP) || defined(CONFIG_IA64) )
 int expand_upwards(struct vm_area_struct *vma, unsigned long address);
 #endif
 
-- 
2.25.1

